### PR TITLE
Move HUD and HUD overlay rendering out of display plugin

### PR DESCRIPTION
**Pull request by SamGondelman**
Instead of passing the HUD overlays to the display plugin to render after the HUD sphere, get a lambda from the display plugin that contains the instructions on how to render the HUD, in desktop or HMD.  Add a render job for rendering the HUD, and add a new overlay layer for drawing in the HUD that gets drawn after that.

Test Plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/1e1a6a4268549147588e4887d46aa53d/raw/e40a96c96945f604c968f61b1e088846f9138218/LayeredTest.js).  You should see 10 overlays appear in front of you (there are actually 12 but two are missing because layering doesn't seem to work on model overlays, I will put up a bug for that separately because it is an issue in master, too).
- The red overlays are normal.  If you stand behind an object, you won't be able to see them.  They should render behind the HUD.
- The green overlays are drawInFront.  If you stand behind an object, you should still see them.  They should render behind the HUD.
- The blue overlays are drawHUDLayer.  If you stand behind an object, you should still see them.  You should also be able to see them on top of the HUD.
- The HUD should behave the same as it does in master, in desktop and HMD.  Turn on the preview and try resizing the window and switching between desktop and HMD.  You shouldn't crash or have any weird colorful flashes on your screen.
- The hand lasers on the HUD should draw properly.  They're broken on the stats window right now (because of https://github.com/highfidelity/hifi/pull/11509), but you should be able to test them on the Developer -> Debug Logs window.
- The mouse pointer should still render on top of everything.

### CODE REVIEW

**Comment by SamGondelman** [comment]
Hm...HUD sphere intersection with the hand laser is broken in master and this PR, which makes testing this tricky.  Hold off on QA until I fix that.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -5053,6 +5053,7 @@ void Application::update(float deltaTime) {
 
         float sensorToWorldScale = getMyAvatar()->getSensorToWorldScale();
         appRenderArgs._sensorToWorldScale = sensorToWorldScale;
+        appRenderArgs._sensorToWorld = getMyAvatar()->getSensorToWorldMatrix();
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: This wasn't being set.

File: interface/src/ui/overlays/Base3DOverlay.cpp [hunk 2]
```diff
@@ -282,8 +290,6 @@ void Base3DOverlay::update(float duration) {
                 }
             });
             scene->enqueueTransaction(transaction);
-        } else {
-            setRenderTransform(latestTransform);
```
Comment (SamGondelman) [inline interface/src/ui/overlays/Base3DOverlay.cpp]: All overlays are properly in the scene now, so this isn't necessary anymore.

File: interface/src/ui/overlays/OverlaysPayload.cpp [hunk 3]
```diff
@@ -49,20 +50,17 @@ namespace render {
...
-        const int LAYER_3D_FRONT = 1;
-        const int LAYER_3D = 0;
-
         if (overlay->is3D()) {
             auto overlay3D = std::dynamic_pointer_cast<Base3DOverlay>(overlay);
             if (overlay3D->getDrawInFront()) {
-                return LAYER_3D_FRONT;
+                return Item::LAYER_3D_FRONT;
+            } else if (overlay3D->getDrawHUDLayer()) {
+                return Item::LAYER_3D_HUD;
```
Comment (SamGondelman) [inline interface/src/ui/overlays/OverlaysPayload.cpp]: If something is marked as drawInFront and drawHUDLayer, I'm not sure what should happen.  Right now it gives drawInFront precedence.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3951/HighFidelity-Beta-PR11542-b4a3ab920450c36949cde73a54d377e307251d39.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Run this. You should see 10 overlays appear in front of you (there are actually 12 but two are missing
because layering doesn't seem to work on model overlays, I will put up a bug for that separately because it is an issue in master, too). **Check**
The red overlays are normal. If you stand behind an object, you won't be able to see them. They should render behind the HUD. **Checked with toolbar**
The green overlays are drawInFront. If you stand behind an object, you should still see them. They should render behind the HUD. **Check**
The blue overlays are drawHUDLayer. If you stand behind an object, you should still see them. You should also be able to see them on top of the HUD. **Check**
The HUD should behave the same as it does in master, in desktop and HMD. Turn on the preview and try resizing the window and switching between desktop and HMD. You shouldn't crash or have any weird colorful flashes on your screen. **Check**
The hand lasers on the HUD should draw properly. They're broken on the stats window right now (because of #11509), but you should be able to test them on the Developer -> Debug Logs window. **Check**
The mouse pointer should still render on top of everything. **Check**

Looks like a solid **Pass** to me :)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3953/HighFidelity-Beta-PR11542-eb23e7df0858d99beb52e19224403ccb6affd77f.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3964/HighFidelity-Beta-PR11542-30e5721a0a26be9e7870f1956d54b0211dc943e8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3968/HighFidelity-Beta-PR11542-adc55f7da60af9baaf6a7738caa1f958f9b6b43f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3968/HighFidelity-Beta-PR11542-adc55f7da60af9baaf6a7738caa1f958f9b6b43f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3971/HighFidelity-Beta-PR11542-3acbd7820b19caa8676f4555fff9fd79bd9ca625.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3971/HighFidelity-Beta-PR11542-3acbd7820b19caa8676f4555fff9fd79bd9ca625.exe

**Pull request merged**